### PR TITLE
docs(root): state the build check-types actually needs

### PR DESCRIPTION
**Pull request by mobeenabdullah**
`AGENTS.md` told contributors that `check-types` and `lint` both run without a build. **Half of that is true**, and the false half fails in the way that costs the most time: it looks like a broken machine.

## Measured on a clean checkout, not argued

| command | result |
|---|---|
| `pnpm lint` | **22 of 22 successful, exit 0** — genuinely build-free |
| `pnpm check-types` | **exit 1** |

The `check-types` failure is `@nextlyhq/admin`, unable to resolve `@nextlyhq/ui`, `nextly/config`, `nextly/field-catalog` and `nextly/api/versions-diff` until their `dist` exists — **438 unresolved imports**.

**19 of 21 packages pass**, which is what makes this expensive: it reads as one package being broken rather than as a missing prerequisite. CI has never caught it because CI builds first.

## The fix is named and verified, not asserted

```
pnpm --filter <pkg>^... build     # ^... is its dependencies, not the package
```

Verified on the failing package rather than documented from memory:

| | admin `check-types` |
|---|---|
| before | 438 errors, exit 1 |
| after that command | **0 errors, exit 0** |

The entry also says which signal separates the two cases, since that is the part that gets misread: a `TS2307` naming a **workspace** package (`nextly/...`, `@nextlyhq/...`) is a missing build, not a defect in the code being checked. That is the same costume `.claude/rules/verifying-merged-work.md` already warns about; this connects the rule to the command that resolves it.

## Why the obvious repair is not general

Recorded so the next person does not spend the afternoon I did.

`packages/admin` already maps `nextly` to `../nextly/src` with tsconfig `paths`. That resolves the **bare** specifier from source and leaves its **subpaths** and `@nextlyhq/ui` still unresolved — a half-measure that looks like a solved problem.

Extending it does not work either. Mapping a sibling's source pulls that sibling's whole program in, and under pnpm isolation the sibling's own devDependencies are not visible from the consumer. Applied to `builder` → `blocks-engine`, `TS2307` was traded for four unrelated `TS7016`/`TS7006` from the engine's untyped `css-tree` imports. Both are the check failing for a reason that is not about the code being checked, so it was reverted.

## Scope

Documentation only, so no changeset. This is the honest-minimum half of a two-part fix: a follow-up in `blocks-engine` can give `css-tree` a declaration so the source mapping becomes viable, which is the real repair. This one lands now so the guide stops telling contributors the opposite of what will happen.

### CODE REVIEW

**Comment by mobeenabdullah** [comment]
@codex please review this PR

File: AGENTS.md [hunk 1]
```diff
@@ -57,8 +57,35 @@ Before editing a package, read its README.md and check for a nested AGENTS.md.
...
-- `pnpm check-types` and `pnpm lint` do NOT need a build first. `pnpm test`
-  does (turbo handles it when run from the root).
+- `pnpm lint` does NOT need a build first. `pnpm test` does (turbo handles it
+  when run from the root).
+- `pnpm check-types` needs one for any package that imports a workspace sibling
+  through its package exports, and that is not a corner case: on a clean
+  checkout the root `pnpm check-types` EXITS 1, because `@nextlyhq/admin` cannot
+  resolve `@nextlyhq/ui`, `nextly/config` or `nextly/field-catalog` until their
+  `dist` exists. Nineteen of twenty-one packages pass, so the failure looks
+  local to one package rather than like a missing prerequisite.
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply AGENTS.md]: You are right and my number was wrong for a reason worth recording: **it came from turbo's cache.**

Re-measured on a tree with every `dist` removed and `--force` so nothing was replayed:

```
pnpm turbo run check-types --continue --force
Tasks:  6 successful, 21 total
Cached: 0 cached, 21 total          exit 2
```

18 packages report `TS2307`, not one. Top unresolved: `@nextlyhq/ui` (384), `@nextlyhq/adapter-drizzle/types` (208), `@nextlyhq/adapter-drizzle` (117), `nextly/config` (68), `@nextlyhq/blocks-engine` (48).

**Checking your finding also caught a second wrong claim you did not raise.** I had written that `lint` was genuinely build-free at 22 of 22. That was cached too. Forced:

```
pnpm turbo run lint --continue --force
Tasks:  8 successful, 22 total       exit 1
```

It fails through `import-x/no-unresolved`, which resolves the same specifiers, so an unbuilt sibling is an unresolved import there as well. **Both halves of the original AGENTS.md sentence were false, not one** — which was the entire premise of this PR.

The entry now recommends the root build, and says to measure these with `--force`, citing this mistake as the reason.

File: AGENTS.md [hunk 2]
```diff
@@ -57,8 +57,35 @@ Before editing a package, read its README.md and check for a nested AGENTS.md.
...
+
+  The specifier tells you which case you are in. `TS2307` naming a workspace
+  package (`nextly/...`, `@nextlyhq/...`) is a missing build, not a defect in
+  the code being checked — the same costume described in
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply AGENTS.md]: Agreed, and the categorical phrasing was the defect. A misspelled specifier, a removed export-map subpath and a broken tsconfig mapping all produce `TS2307` and survive any rebuild, so telling a reader to treat it as environmental sends them away from a real regression.

Rewritten: a workspace `TS2307` is now the USUAL diagnosis that **a successful rebuild confirms**, not a reading applied in advance. If the error survives a build of that package's dependencies, it is a real resolution defect. It also now points at `.claude/rules/verifying-merged-work.md` for the "check what `main` changed first" step rather than only borrowing its costume metaphor.

File: AGENTS.md [hunk 3]
```diff
@@ -57,8 +57,35 @@ Before editing a package, read its README.md and check for a nested AGENTS.md.
...
+  A package importing no workspace sibling is genuinely build-free, which is why
+  the guarantee held for so long and why it fails without warning the first time
+  a package takes such a dependency.
+
+  `packages/admin` shows the half-measure that does not close this: it maps
+  `nextly` to `../nextly/src` with tsconfig `paths`, so the bare specifier
+  resolves from source, while its SUBPATHS and `@nextlyhq/ui` still do not.
+  Mapping a sibling's source is also not a general fix — it pulls that sibling's
+  whole program in, and under pnpm isolation its own devDependencies are not
+  visible from the consumer, so the errors change rather than stop.
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply AGENTS.md]: Removed, because I could not stand it up. The pnpm-isolation rationale was carried over from a task note written in an earlier session rather than measured here, and a claim I did not verify has no business in the contributor guide.

What the run actually shows is narrower and checkable: `packages/admin` maps the bare `nextly` specifier to `../nextly/src` and resolves it without a build, while `packages/plugin-sdk` has no such mapping and reports `Cannot find module 'nextly'`. The subpaths — `nextly/config`, `nextly/field-catalog` — are not covered by that mapping and still need `dist`. So path mappings cover part of this and are not a general answer, which is the useful point; the mechanism I gave for why extending them fails is gone rather than restated.

**Comment by mobeenabdullah** [comment]
@codex please review this PR

File: AGENTS.md [hunk 4]
```diff
@@ -57,8 +57,45 @@ Before editing a package, read its README.md and check for a nested AGENTS.md.
...
 - `pnpm build` builds all packages (turbo, dependency order).
-- `pnpm check-types` and `pnpm lint` do NOT need a build first. `pnpm test`
-  does (turbo handles it when run from the root).
+- `pnpm check-types` and `pnpm lint` BOTH need a build first. On a clean
+  checkout, measured with the cache forced off so the numbers are of work that
+  actually ran:
+
+  | command                                         | result on a clean tree                             |
+  | ----------------------------------------------- | -------------------------------------------------- |
+  | `pnpm turbo run check-types --continue --force` | 6 of 21 successful, 18 packages reporting `TS2307` |
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply AGENTS.md]: Corrected to 15, and you are right about where mine came from. Turbo's own `Failed:` line names 15 packages; my 18 came from a grep whose `-B1` context bled prefixes from neighbouring packages into the count.

That is the second number in this entry produced by a sloppy measurement rather than read off the tool — the first was the cached run. The entry now takes its figure from turbo's own accounting.

File: AGENTS.md [hunk 5]
```diff
@@ -57,8 +57,45 @@ Before editing a package, read its README.md and check for a nested AGENTS.md.
...
+  | `pnpm turbo run lint --continue --force`        | 8 of 22 successful                                 |
+
+  `check-types` fails because a workspace import resolves through the sibling's
+  package exports to a `dist/index.d.ts` that does not exist yet. `lint` fails
+  for the same underlying reason through a different rule: `import-x/no-unresolved`
+  resolves the same specifiers, so an unbuilt sibling is an unresolved import.
+
+  Run `pnpm build` first. A per-package `pnpm --filter <pkg>^... build` builds
+  only that package's dependencies and is enough when you are checking one
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply AGENTS.md]: Confirmed against `pnpm recursive --help`, which defines `<pattern>^...` as the dependencies "without including the matched packages". So the package's own `dist` stays absent and `lint` fails on its self-imports — the same missing build wearing a different rule's error message, which is exactly the confusion this entry exists to prevent.

Changed to `pnpm --filter <pkg>... build`, with the trailing form and the reason named inline so the next reader does not reintroduce the caret. The whole-repo build is now the stated default, since the failure is workspace-wide rather than local.

**Comment by mobeenabdullah** [comment]
@codex please review this PR

File: AGENTS.md [hunk 6]
```diff
@@ -57,8 +57,55 @@ Before editing a package, read its README.md and check for a nested AGENTS.md.
...
+  an earlier built state reports a clean tree as passing. That is not a
+  hypothetical: the first version of this entry claimed 19 of 21 passing and
+  `lint` clean at 22 of 22, and both numbers came from cache hits.
+
+  `TS2307` naming a workspace package (`nextly/...`, `@nextlyhq/...`) is USUALLY
+  a missing build, and the rebuild is what confirms it. It is not proof on its
+  own: a misspelled specifier, a removed export-map subpath, or a broken
+  tsconfig path mapping produces exactly the same error, and no amount of
+  rebuilding fixes those. If the error survives a successful build of that
+  package's dependencies, it is a real resolution defect — see
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply AGENTS.md]: Right, and the inconsistency was mine: I corrected the selector and left the diagnostic still saying "dependencies". For a package with self-imports those can all build while its own `dist` stays absent, so the sentence misclassified precisely the missing-output failure the paragraph above it describes.

It now reads "survives a successful build of that package AND its dependencies — the `<pkg>...` form above, not `^...`".

While fixing it I added the other half, which this entry was missing entirely. Two states distort these numbers in OPPOSITE directions and neither shows in the summary line: **a warm cache overstates health, a missing `dist` overstates breakage.** Both measured on this repository at this commit — 19 of 21 passing from cache, and 361 unresolved `@nextlyhq/ui` imports with vitest unable to collect on a freshly installed unbuilt tree. The entry now says to state the cache and build state when quoting a figure, because on its own a number from either end is misleading in a different direction.

**Comment by mobeenabdullah** [comment]
@codex please review this PR
